bundles/io.openliberty.tools.eclipse.ui/src/io/openliberty/tools/eclipse/WorkspaceProjectsModel.java

```diff
@@ -0,0 +1,197 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial implementation
+ *******************************************************************************/
+package io.openliberty.tools.eclipse;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IWorkspaceRoot;
+import org.eclipse.core.resources.ResourcesPlugin;
+
+import io.openliberty.tools.eclipse.logging.Trace;
+import io.openliberty.tools.eclipse.utils.ErrorHandler;
+
+/**
+ * Represents the project model informing the Liberty tools dashboard and the Run Configurations
+ */
+public class WorkspaceProjectsModel {
+
+    private Map<String, Project> projectsByLocation;
+    private Map<String, Project> projectsByName;
+
+    /**
+     * Constructor.
+     */
+    public WorkspaceProjectsModel() {
+        initProjectModels();
+    }
+
+    private void createProjectModels(List<IProject> projects, boolean classify) {
```

Comment:
This method is given a list of all open projects in the workspace. They can be Liberty projects or not. This method does not filter Liberty projects, so the dashboard would end up with every project in the workspace (Liberty or not). Also since refresh is associated with this call, that would also be broken.

Reply:
The public "get/read"-type methods of `WorkspaceProjectsModel` only include: